pkg/plugins/common/kustomize/v1/init.go

```diff
@@ -94,6 +108,30 @@ func (p *initSubcommand) InjectConfig(c config.Config) error {
 	return nil
 }
 
+func (p *initSubcommand) PreScaffold(machinery.Filesystem) error {
+	arch := runtime.GOARCH
+	arch = strings.Replace(arch, "x86_64", "amd64", -1)
```

Comment:
Is this replace necessary? I don't think `x86_64` is a valid value for `GOARCH` so `runtime.GOARCH` should default a `x86_64` system to a value of `amd64`.

For reference, if I run this Go code:
```go
package main

import (
	"fmt"
	"runtime"
)

func main() {
	fmt.Println("PLATFORM:", runtime.GOOS+"/"+runtime.GOARCH)
}
```
with the following command:
```
go run main.go
```
I get this output (on a x86_64 arch):
```
PLATFORM: linux/amd64
```

However if I run it with:
```
GOARCH=x86_64 go run main.go
```
I get the following output:
```
cmd/go: unsupported GOOS/GOARCH pair linux/x86_64
```

This makes me think that we should not support the case where GOARCH is set to be `x86_64` because then a user is explicitly setting their Go environment to be invalid by setting GOARCH to an invalid value.

WDYT?

Reply:
Another thing to note, when running `go tool dist list` the output does not contain any `x86_64` and the output is:
```
aix/ppc64
android/386
android/amd64
android/arm
android/arm64
darwin/amd64
darwin/arm64
dragonfly/amd64
freebsd/386
freebsd/amd64
freebsd/arm
freebsd/arm64
illumos/amd64
ios/amd64
ios/arm64
js/wasm
linux/386
linux/amd64
linux/arm
linux/arm64
linux/mips
linux/mips64
linux/mips64le
linux/mipsle
linux/ppc64
linux/ppc64le
linux/riscv64
linux/s390x
netbsd/386
netbsd/amd64
netbsd/arm
netbsd/arm64
openbsd/386
openbsd/amd64
openbsd/arm
openbsd/arm64
openbsd/mips64
plan9/386
plan9/amd64
plan9/arm
solaris/amd64
windows/386
windows/amd64
windows/arm
windows/arm64
```

Reply:
the support here is regarding the binaries (have or not the bin to download)
If you check our makefile target, we are replacing linux/x86_64 by linux/amd64 because this platform is built with this name, so I'd prefer to keep the consistence.

Reply:
@camilamacedo86 this looks good, feel free to mark it as resolved